clustered/client/src/test/java/org/ehcache/clustered/client/internal/config/xml/ClusteringServiceConfigurationParserTest.java

```diff
@@ -424,6 +443,227 @@ public void testServersWithClusterTierManagerAndOptionalPorts() throws Exception
     assertThat(servers, is(expectedServers));
   }
 
+  @Test
+  public void testTranslateServiceCreationConfiguration() throws Exception {
+    URI connectionUri = new URI("terracotta://localhost:9510/my-application");
+    ClusteringServiceConfiguration serviceConfig = ClusteringServiceConfigurationBuilder.cluster(connectionUri)
+      .timeouts(Timeouts.DEFAULT)
+      .autoCreate()
+      .defaultServerResource("main")
+      .resourcePool("primaryresource", 5, MemoryUnit.GB)
+      .resourcePool("seondaryresource", 10, MemoryUnit.GB, "optional")
+      .build();
+    Configuration clusterConfiguration = getClusterConfiguration(serviceConfig);
+    ConfigurationTranslator configurationTranslator = new ConfigurationTranslator(clusterConfiguration);
+    String retValue = configurationTranslator.toXml();
+    System.out.println(retValue);
+    String startTag = "<ehcache:config xmlns:ehcache=\"http://www.ehcache.org/v3\"><ehcache:service>";
+    String endTag = "</ehcache:service></ehcache:config>";
+    String entireString = startTag + retValue + endTag;
+    System.out.println(entireString);
+    XmlConfiguration config = new XmlConfiguration(makeConfig(new String[] { entireString }));
+
+    assertThat(config.getServiceCreationConfigurations(), is(notNullValue()));
+    assertThat(config.getServiceCreationConfigurations(), iterableWithSize(equalTo(1)));
+
+    ClusteringServiceConfiguration retServiceConfig =
+      (ClusteringServiceConfiguration)config.getServiceCreationConfigurations().iterator().next();
+
+    assertThat(connectionUri, equalTo(retServiceConfig.getClusterUri()));
+    assertThat(true, is(retServiceConfig.isAutoCreate()));
+    assertThat(retServiceConfig.getServerConfiguration(), is(notNullValue()));
+
+    assertThat(retServiceConfig.getTimeouts(), is(notNullValue()));
+    assertThat(retServiceConfig.getTimeouts().getConnectionTimeout(), equalTo(Timeouts.DEFAULT.getConnectionTimeout()));
+    assertThat(retServiceConfig.getTimeouts()
+      .getReadOperationTimeout(), equalTo(Timeouts.DEFAULT.getReadOperationTimeout()));
+    assertThat(retServiceConfig.getTimeouts().getConnectionTimeout(), equalTo(Timeouts.DEFAULT.getConnectionTimeout()));
+
+    Map<String, ServerSideConfiguration.Pool> retPoolMap = retServiceConfig.getServerConfiguration().getResourcePools();
+
+    assertThat(retPoolMap, is(notNullValue()));
+    assertThat(retPoolMap.size(), is(equalTo(2)));
+
+    for (Map.Entry<String, ServerSideConfiguration.Pool> entry : retPoolMap.entrySet()) {
+      String name = entry.getKey();
+      ServerSideConfiguration.Pool value = entry.getValue();
+      assertThat(("primaryresource".equals(name) || "seondaryresource".equals(name)), is(true));
+      if ("primaryresource".equals(name)) {
+        assertThat(5 * 1024 * 1024 * 1024L, is(equalTo(value.getSize())));
```

Comment:
assertj? :-)

Reply:
:)
I mistakenly used assertThat from hamcrest, originally the assert used in this file was org.junit.Assert.assertThat. I am bringing back the old assertThat.

With respect to assertj,  if we want to use assertj, then imo, that should be used for entire file (i.e. for the older existing tests as well). Do you want to me to use assertj for the entire file? If yes, then can we take it as a different task as that will be a general refactoring?